esrally/telemetry.py

```diff
@@ -568,6 +568,138 @@ def record(self):
                 self.metrics_store.put_doc(doc, level=MetaInfoScope.cluster, meta_data=shard_metadata)
 
 
+class ShardStats(TelemetryDevice):
+    """
+    Collects and pushes shard stats for the specified cluster to the metric store.
+    """
+    internal = False
+    command = "shard-stats"
+    human_name = "Shard Stats"
+    help = "Regularly samples nodes stats at shard level"
+
+    def __init__(self, telemetry_params, clients, metrics_store):
+        """
+        :param telemetry_params: The configuration object for telemetry_params.
+            May optionally specify:
+            ``shard-stats-indices``: JSON structure specifying the index pattern per cluster to publish stats from.
+            Not all clusters need to be specified, but any name used must be be present in target.hosts. Alternatively,
+            the index pattern can be specified as a string can be specified in case only one cluster is involved.
+            Example:
+            {"shard-stats-indices": {"cluster_a": ["follower"],"default": ["leader"]}
+
+            ``shard-stats-sample-interval``: positive integer controlling the sampling interval. Default: 60 seconds.
+        :param clients: A dict of clients to all clusters.
+        :param metrics_store: The configured metrics store we write to.
+        """
+        super().__init__()
+
+        self.telemetry_params = telemetry_params
+        self.clients = clients
+        self.sample_interval = telemetry_params.get("shard-stats-sample-interval", 60)
+        if self.sample_interval <= 0:
+            raise exceptions.SystemSetupError(
+                f"The telemetry parameter 'shard-stats-sample-interval' must be greater than zero but was {self.sample_interval}.")
+
+        self.specified_cluster_names = self.clients.keys()
+        indices_per_cluster = self.telemetry_params.get("shard-stats-indices", False)
```

Comment:
is `False` a valid value here? Shouldn't this be a * by default i.e. we monitor all indices? 
Its not clear what happens if nothing is specified for `shard-stats-indices` given `specified_cluster_names` will not be set.

Reply:
maybe `{opts.TargetHosts.DEFAULT: "*"}` as the default.

Reply:
So, this was leftover from when i used _cat api and there was a way to pass index parameter to the query to only get stats for specific indices. Looks like nodes api in elasticsearch py client doesn't allow that. I can still keep and process on rally side if we think it could be useful. thoughts?

Reply:
We don't expect that there are many indices in the cluster that are not created because of the benchmark so I'd opt for simplicity and remove the parameter.

Reply:
agreed any non-benchmark indices will be negligible. I'd like to filter out internal and system indices but I think we can do that in our dashboarding.